### DISCUSSION TITLE
How to resolve OKD vulnerabilities?

### DISCUSSION THREAD

**Discussion by ALL-SPACE-Anas** [okd-project/okd#2065] [Q&A] [answered]
Hi,

I wanted to install "Red Hat Quay Container Security Operator" and "Compliance Operator" but from the discussion https://github.com/okd-project/okd/discussions/2046 I found that it's not possible to do that in OKD.

I wanted to have a system to monitor and scan containers as well as nodes for any vulnerabilities. I found and installed neuvector operator which seems to work nicely. It shows a number of node-level vulnerabilities (basically package updates). My question is, do I need to wait for a minor update to OKD 4.15 for these to resolve or can I somehow resolve these myself?

**Comment by GingerGeek** [comment, marked as answer]
Hey, thanks for getting in touch.

Node package updates would generally be applied as part of an overall cluster update, as the Node OS state is part of an OKD release payload.

We are about to release 4.16/4.17 that contain more updated OS images which should quell most vulnerable packages.

If you need critical security fixes, you could manually layer them on top of the node OS. [Here are the docs for that](https://docs.okd.io/latest/machine_configuration/mco-coreos-layering.html).

**Comment by ALL-SPACE-Anas** [comment]
Hi @GingerGeek, thanks for responding to this and the very helpful link. I guess I'll wait for the updates to be released. I'm assuming the updates would show up on the `stable-4` update channel. Please let me know if I need to move to a different update channel.

Thanks
